tests/DependencyInjection/Compiler/AddProvidersPassTest.php

```diff
@@ -84,4 +87,50 @@ public function testProcessForManyProviders()
         $providersPass = new AddProvidersPass();
         $providersPass->process($containerBuilderMock);
     }
+
+    public function testPriorityRegisteredProviders()
+    {
+        $expectedProviders = [];
+        $expectedProviders[0][] = new Reference('id');
+        $expectedProviders[20][] = new Reference('id2');
+
+        krsort($expectedProviders);
+        $expectedProviders = call_user_func_array('array_merge', $expectedProviders);
+
+        if (class_exists(IteratorArgument::class)) {
+            $expectedProviders = new IteratorArgument($expectedProviders);
+        }
+
+        $definitionMock = $this->getMockBuilder('Symfony\Component\DependencyInjection\Definition')
+            ->disableOriginalConstructor()
```

Comment:
you can use `$this->createMock(Class)` instead of getMockBuilder->disableOriginalConstructor->getMock (at least i think so, not entirely sure if PHP 5.6 will install a too old version of phpunit that is missing this feature.